src/runtime/proxy-component.ts

```diff
@@ -133,25 +133,60 @@ export const proxyComponent = (
             // if the propName exists on the prototype of `Cstr`, this update may be a result of Stencil using native
             // APIs to reflect props as attributes. Calls to `setAttribute(someElement, propName)` will result in
             // `propName` to be converted to a `DOMString`, which may not be what we want for other primitive props.
+            return;
+          } else if (propName == null) {
```

Comment:
Most of the stuff in this block comes from the existing [`setValue()`](https://github.com/ionic-team/stencil/blob/main/src/runtime/set-value.ts#L11C2-L11C2) function